### PR TITLE
Concurrent requests fail

### LINKED ISSUE #22: Repeated error messages
(Probably a caching issue, but not confirmed)

Users have reported receiving error messages when attempting to download stories. The issue is inconsistent and does not always happen. So far the error messages I've personally confirmed (for stories that definitely exist) include:

- Internal server error (completely unhandled exception)
- ClientResponseError 429 (Rate-limit by Wattpad)
- ClientResponseError 400|404 (This story does not exist)

Relevant logs:
```
INFO:     10.0.1.182:52690 - "GET /download/314175600?om=1&download_images=true&mode=story HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/usr/local/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/applications.py", line 116, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 55, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 44, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 746, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 75, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 55, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 44, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 70, in app
    response = await func(request)
  File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 299, in app
    raise e
  File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 294, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/app/main.py", line 103, in handle_download
    async for title in add_chapters(
  File "/app/create_book.py", line 215, in add_chapters
    async with session.get(image["src"]) as response:
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 1187, in __aenter__
    self._resp = await self._coro
  File "/usr/local/lib/python3.10/site-packages/aiohttp_client_cache/session.py", line 51, in _request
    response, actions = await self.cache.request(
  File "/usr/local/lib/python3.10/site-packages/aiohttp_client_cache/backends/base.py", line 139, in request
    response = None if actions.skip_read else await self.get_response(actions.key)
  File "/usr/local/lib/python3.10/site-packages/aiohttp_client_cache/backends/base.py", line 147, in get_response
    response = await self.responses.read(key) or await self._get_redirect_response(str(key))
  File "/usr/local/lib/python3.10/site-packages/aiohttp_client_cache/backends/base.py", line 169, in _get_redirect_response
    redirect_key = await self.redirects.read(key)
  File "/usr/local/lib/python3.10/site-packages/aiohttp_client_cache/backends/sqlite.py", line 180, in read
    row = await cursor.fetchone()
  File "/usr/local/lib/python3.10/site-packages/aiosqlite/cursor.py", line 65, in fetchone
    return await self._execute(self._cursor.fetchone)
  File "/usr/local/lib/python3.10/site-packages/aiosqlite/cursor.py", line 40, in _execute
    return await self._conn._execute(fn, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/aiosqlite/core.py", line 133, in _execute
    return await future
  File "/usr/local/lib/python3.10/site-packages/aiosqlite/core.py", line 106, in run
    result = function()
sqlite3.ProgrammingError: Cannot operate on a closed database.
```

### PR DESCRIPTION
**Pull request by TheOnlyWayUp**
Fixes cache database locking on concurrent requests by adding an option to use Redis as a cache backend.

This PR also adds Eliot logging (`DEBUG=1 python3 main.py`), Request cancellation when the Client disconnects, and standardizes the usage of CachedSession.

Fixes #22

### CODE REVIEW

**Comment by AaronBenDaniel** [comment]
`Eliot` is not included in `/src/api/requirements.txt`

**Comment by AaronBenDaniel** [comment]
`pydantic_settings` is not included in `/src/api/requirements.txt`

File: src/api/requirements.txt
Comment (AaronBenDaniel) [inline src/api/requirements.txt:1]: `eliot` and `pydantic_settings` are not included, image fails to run without them

File: src/api/src/main.py [hunk 1]
```diff
@@ -69,60 +110,72 @@ async def handle_download(
...
+            cookies = None
+
+        match mode:
+            case DownloadMode.story:
+                story_id = download_id
+                metadata = await retrieve_story(story_id, cookies)
+            case DownloadMode.part:
+                story_id, metadata = await fetch_story_id(download_id, cookies)
+
+        logger.error(f"Retrieved story id ({story_id=})")
```
Comment (AaronBenDaniel) [inline src/api/src/main.py:147]: Shouldn't this be `logger.info` instead of `logger.error`?

Comment (TheOnlyWayUp) [reply src/api/src/main.py]: Oh, that's right

**Review by AaronBenDaniel — APPROVED** [review]

**Comment by TheOnlyWayUp** [comment]
<3

**Pull request merged**